_rules/image-not-in-acc-tree-is-decorative-e88epe.md

```diff
@@ -0,0 +1,300 @@
+name: Image not in the accessibility tree is decorative
+rule_type: atomic
+description: |
+  This rule checks that visible img, svg and canvas elements that are ignored by assistive technologies are decorative
+accessibility_requirements:
+  wcag20:1.1.1: # Non-text content (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - Accessibiliy tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Wilco Fiers
+  assets:
+    - W3C
+    - Wikimedia
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'attr-lowercase'
+---
+
+## Applicability
+
+Any `img`, `canvas` or `svg` element that is [visible][] and is not [included in the accessibility tree][], except if one of the following is true:
+
+- **unloaded img**: the element is an `img` and its [current request][]'s [state][image request state] is not [completely available][]; or
+- **ignored svg**: the element is an `svg` with an empty (`""`) [accessible name][] and a [semantic role][] of `graphics-document`; or
+- **ignored canvas**: the element is a `canvas` with an empty (`""`) [accessible name][] and no [explicit semantic role][]; or
+- **named from author**: the element is a [descendant][] in the [flat tree][] of an element that is [named from author][].
+
+**Note**: An example of an image ignored because of "named from author" is when the image is a descendant of a `button` element that uses `aria-label` for its accessible name.
+
+## Expectation
+
+Each test target is [purely decorative][].
+
+**Note**: It is relatively common for an informative image such as an icon to be marked up as decorative, if the text alternative is adjacent to the image. This is a [conforming alternative version][] for the image. This fails the rule but meets [conformance requirement 1 of WCAG 2.1](https://www.w3.org/TR/WCAG21/#cc1).
+
+## Assumptions
+
+- `svg` elements with a [semantic role][] of `graphics-document` and with an empty (`""`) [accessible name][] are ignored by assistive technologies tested for this rule. If some assistive technology does not ignore these elements, and that assistive technology is required for conformance, passing this rule does not ensure all decorative `svg` elements can be ignored, and the [success criterion 1.1.1 Non-text content][] may still not be satisfied. The same is true for `canvas` elements with no [semantic role][] and an empty (`""`) [accessible name][].
+
+- A web page with informative images without an [accessible name][] may conform to WCAG 2.1 Level A when the information provided by that image is available elsewhere, either on the web page itself or on a [conforming alternative version][] of the page. For example if an equivalent text is adjacent to the image, or if the text alternative is included in the [accessible name][] of a parent element.
```

Comment:
As a general note (not just for this rule), I'd like assumptions to state clearly and precisely (as unambiguously as possible) what is assumed, before explaining the reasons or consequence for it (we haven't really standardised how assumptions should look like…)

The reason for this is that I believe it would make testing more transparent. If a rule fails but the page conforms to WCAG, we need to be able to precisely pinpoint whether "this assumption is false for this page, so the rule cannot be applied" or "oops, there is a bug in the rule". Having a clear and precise statement of the assumption will help testers reach the same conclusion.

---
In this case, I'd suggest:
```suggestion
## Assumptions

- This rule assumes that `svg` elements with a [semantic role][] of `graphics-document` and with an empty (`""`) [accessible name][] are ignored by assistive technologies. This is the case for all assistive technologies tested for this rule. If some assistive technology does not ignore these elements, and that assistive technology is required for conformance, passing this rule does not ensure all decorative `svg` elements can be ignored, and the [success criterion 1.1.1 Non-text content][] may still not be satisfied. The same is true for `canvas` elements with no [semantic role][] and an empty (`""`) [accessible name][].

- This rule assumes that there is no [conforming alternative version][] of the page or the image being tested. A web page with informative images without an [accessible name][] may conform to WCAG 2.1 Level A when the information provided by that image is available elsewhere, either on the web page itself or on a [conforming alternative version][] of the page. For example if an equivalent text is adjacent to the image, or if the text alternative is included in the [accessible name][] of a parent element.
```

Reply:
I disagree. The "Assumption" heading should be sufficient for that in my opinion.